### DISCUSSION TITLE
Postprocessing: Total mass and total energy calculation porous flow module

### DISCUSSION THREAD

**Discussion by rksin8** [idaholab/moose#18558] [Q&A Modules: General] [answered]
Hi,
I am solving a single-phase TH problem(porous_flow module)  where boundary conditions on a face are Dirichlet type for both pressure and temperature. How do I calculate the total mass and total energy for the given boundary? 

Thank you,
Ranjeet

**Comment by WilkAndy** [comment, marked as answer]
The simplest situation is when you're interested in the total flow into (or out of) the entire system.  Then you can use [PorousFlowFluidMass](https://mooseframework.inl.gov/source/postprocessors/PorousFlowFluidMass.html) or [PorousFlowHeatEnergy](https://mooseframework.inl.gov/source/postprocessors/PorousFlowHeatEnergy.html).

Note that Dirichlet conditions are [not recommended in PorousFlow](https://mooseframework.inl.gov/modules/porous_flow/boundaries.html) since they frequently end up trying to withdraw fluid or heat from a region which contains none, so we try to encourage people to use one of the many [PorousFlowSinks](https://mooseframework.inl.gov/source/bcs/PorousFlowSink.html) instead.

Nevertheless, Dirichlet BCs are really convenient.  So, use a [PorousFlowOutflowBC](https://mooseframework.inl.gov/source/bcs/PorousFlowOutflowBC.html) to record your flux (study the `save_in` feature on that webpage).

**Reply by rksin8** [reply]
@WilkAndy Thank you for the explanation. Calculated injected and produced mass using the Save-flux feature in these Boundary conditions.
